contracts/flow/token/BloctoToken.cdc

```diff
@@ -1,44 +1,125 @@
-    pub event TokensMinted(amount: UFix64)
-
+    access(all)
+    event TokensMinted(amount: UFix64)
+    
     // Event that is emitted when tokens are destroyed
-    pub event TokensBurned(amount: UFix64)
-
+    access(all)
+    event TokensBurned(amount: UFix64)
+    
     // Event that is emitted when a new minter resource is created
-    pub event MinterCreated(allowedAmount: UFix64)
-
+    access(all)
+    event MinterCreated(allowedAmount: UFix64)
+    
     // Event that is emitted when a new burner resource is created
-    pub event BurnerCreated()
+    access(all)
+    event BurnerCreated()
+    
+
+     // Gets a list of the metadata views that this contract supports
+    access(all) view fun getContractViews(resourceType: Type?): [Type] {
+        return [
+            Type<FungibleTokenMetadataViews.FTView>(),
+            Type<FungibleTokenMetadataViews.FTDisplay>(),
+            Type<FungibleTokenMetadataViews.FTVaultData>(),
+            Type<FungibleTokenMetadataViews.TotalSupply>()
+            ]
+    }
 
+    /// Get a Metadata View from BloctoToken
+    ///
+    /// @param view: The Type of the desired view.
+    /// @return A structure representing the requested view.
+    ///
+    access(all) fun resolveContractView(resourceType: Type?, viewType: Type): AnyStruct? {
+        switch viewType {
+            case Type<FungibleTokenMetadataViews.FTView>():
+                return FungibleTokenMetadataViews.FTView(
+                    ftDisplay: self.resolveContractView(resourceType: nil, viewType: Type<FungibleTokenMetadataViews.FTDisplay>()) as! FungibleTokenMetadataViews.FTDisplay?,
+                    ftVaultData: self.resolveContractView(resourceType: nil, viewType: Type<FungibleTokenMetadataViews.FTVaultData>()) as! FungibleTokenMetadataViews.FTVaultData?
+                )
+            case Type<FungibleTokenMetadataViews.FTDisplay>():
+                let media = MetadataViews.Media(
+                    file: MetadataViews.HTTPFile(
+                        url: "https://raw.githubusercontent.com/blocto/assets/main/color/flow/blt.svg"
+                    ),
+                    mediaType: "image/svgxml"
+                )
+                let medias = MetadataViews.Medias([media])
+                return FungibleTokenMetadataViews.FTDisplay(
+                    name: "Blocto Token",
+                    symbol: "BLT",
+                    description: "Blocto token (BLT) is the utility and governance token of Blocto.",
+                    externalURL: MetadataViews.ExternalURL("https://blocto.io/"),
+                    logos: medias,
+                    socials: {
+                        "x": MetadataViews.ExternalURL("https://x.com/BloctoApp")
+                    }
+                )
+            case Type<FungibleTokenMetadataViews.FTVaultData>():
+                let vaultRef = BloctoToken.account.storage.borrow<auth(FungibleToken.Withdraw) &BloctoToken.Vault>(from: /storage/BloctoTokenVault)
```

Comment:
Typo, should be `from: /storage/bloctoTokenVault`
Could use `TokenStoragePath` directly.